### PR TITLE
Azure SQL database support

### PR DESCRIPTION
**Pull request by RobClenshaw**
Here are some minor fixes to support running these tests on Azure SQL databases.

### CODE REVIEW

File: Current/Max degree of parallelism.sql [hunk 1]
```diff
@@ -10,13 +10,24 @@ BEGIN
...
-    Declare @Output VarChar(max)
+    Declare @Output VarChar(max), @Warning VarChar(max)
     Set @Output = ''
+    Set @Warning = 'Warning: Max degree of parallelism is setup to use all cores'
 
-    select @Output = 'Warning: Max degree of parallelism is setup to use all cores'
-    from   sys.configurations
-    where  name = 'max degree of parallelism'
-           and value_in_use = 0
+    IF (SERVERPROPERTY('EngineEdition') = 5)
```
Comment (MichelleTaylorRG) [inline Current/Max degree of parallelism.sql]: Have you checked whether this is also the case on Managed Instances (which have a different EngineEdition)?

File: Current/Performance counters permission check.sql [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+GO
+
+CREATE FUNCTION [SQLCop].[DmOsPerformanceCountersPermissionErrors]()
+RETURNS VARCHAR(MAX)
+AS
+BEGIN
+    DECLARE @Result VARCHAR(MAX)
+    SET @Result = ''
+
+    IF (SERVERPROPERTY('EngineEdition') = 5)
```
Comment (MichelleTaylorRG) [inline Current/Performance counters permission check.sql:12]: See managed instance comment above

File: Current/Max degree of parallelism.sql (hunk 1 above)
Comment (RobClenshaw) [reply Current/Max degree of parallelism.sql]: Not yet, this was just to support Azure SQL Databases.

File: Current/Performance counters permission check.sql (hunk 2 above)
Comment (RobClenshaw) [reply Current/Performance counters permission check.sql:12]: I've done some testing and the permissions required on a managed instance match regular SQL Server so I think this function is ready to go.

**Pull request merged**